public/app/plugins/datasource/elasticsearch/components/QueryEditor/BucketAggregationsEditor/SettingsEditor/DateHistogramSettingsEditor.tsx

```diff
@@ -42,51 +43,31 @@ interface Props {
   bucketAgg: DateHistogram;
 }
 
-const getInitialState = (initialValue?: string): IntervalOption[] => {
-  return defaultIntervalOptions.concat(
-    defaultIntervalOptions.some(hasValue(initialValue))
-      ? []
-      : {
-          value: initialValue,
-          label: initialValue,
-        }
-  );
-};
-
 export const DateHistogramSettingsEditor = ({ bucketAgg }: Props) => {
   const dispatch = useDispatch();
 
-  const [intervalOptions, setIntervalOptions] = useState<IntervalOption[]>(
-    getInitialState(bucketAgg.settings?.interval)
-  );
-
-  const addIntervalOption = (value: string) => setIntervalOptions([...intervalOptions, { value, label: value }]);
-
   const handleIntervalChange = (v: string) => dispatch(changeBucketAggregationSetting(bucketAgg, 'interval', v));
 
   return (
     <>
       <InlineField label="Interval" {...inlineFieldProps}>
-        <Select<string>
+        <Select
           inputId={uniqueId('es-date_histogram-interval')}
-          onChange={(e) => handleIntervalChange(e.value!)}
-          options={intervalOptions}
-          value={bucketAgg.settings?.interval || bucketAggregationConfig[bucketAgg.type].defaultSettings?.interval}
-          allowCustomValue
           isValidNewOption={isValidNewOption}
           filterOption={optionStartsWithValue}
-          onCreateOption={(value) => {
-            addIntervalOption(value);
-            handleIntervalChange(value);
-          }}
+          {...useCreatableSelectPersistedBehaviour({
+            options: defaultIntervalOptions,
+            value: bucketAgg.settings?.interval || bucketAggregationConfig.date_histogram.defaultSettings?.interval,
+            onChange: handleIntervalChange,
+          })}
         />
       </InlineField>
 
       <InlineField label="Min Doc Count" {...inlineFieldProps}>
         <Input
           onBlur={(e) => dispatch(changeBucketAggregationSetting(bucketAgg, 'min_doc_count', e.target.value!))}
           defaultValue={
-            bucketAgg.settings?.min_doc_count || bucketAggregationConfig[bucketAgg.type].defaultSettings?.min_doc_count
+            bucketAgg.settings?.min_doc_count || bucketAggregationConfig.date_histogram.defaultSettings?.min_doc_count
```

Comment:
This is practically the same thing. So I am just curious, why did you decide to change it from `[bucketAgg.type]` to `date_histogram`.

Reply:
because i felt it was going to be more readable. as you said, it can only be `date_histogram` so I don't see the point of using a computed property here.

in my head this the equivalent of doing `const a = 2;` instead of `const a = 1 + 1`: less overhead when reading the code.

but yeah, aside from this there's no practical reason :)

Reply:
btw, open to revert the change if you think is better

Reply:
No I think it makes sense and is more readable. :) Just was curious!